modules/mono/glue/cs_files/Basis.cs

```diff
@@ -347,6 +350,48 @@ public Vector3 xform_inv(Vector3 v)
             );
         }
 
+		public Quat Quat() {
+			float trace = x[0] + y[1] + z[2];
+
+			if (trace > 0.0f) {
+				float s = Mathf.sqrt(trace + 1.0f) * 2f;
+				float inv_s = 1f / s;
+				return new Quat(
+					(z[1] - y[2]) * inv_s,
+					(x[2] - z[0]) * inv_s,
+					(y[0] - x[1]) * inv_s,
+					s * 0.25f
+				);
+			} else if (x[0] > y[1] && x[0] > z[2]) {
+				float s = Mathf.sqrt(x[0] - y[1] - z[2] + 1.0f) * 2f;
+				float inv_s = 1f / s;
+				return new Quat(
+					s * 0.25f,
+					(x[1] + y[0]) * inv_s,
+					(x[2] + z[0]) * inv_s,
+					(z[1] - y[2]) * inv_s
+				);
+			} else if (y[1] > z[2]) {
+				float s = Mathf.sqrt(-x[0] + y[1] - z[2] + 1.0f) * 2f;
+				float inv_s = 1f / s;
+				return new Quat(
+					(x[1] + y[0]) * inv_s,
+					s * 0.25f,
+					(y[2] + z[1]) * inv_s,
+					(x[2] - z[0]) * inv_s
+				);
+			} else {
+				float s = Mathf.sqrt(-x[0] - y[1] + z[2] + 1.0f) * 2f;
+				float inv_s = 1f / s;
+				return new Quat(
+					(x[2] + z[0]) * inv_s,
+					(y[2] + z[1]) * inv_s,
+					s * 0.25f,
+					(y[0] - x[1]) * inv_s
+				);
+			}
+		}
+
```

Comment:
What is this for? I don't see it anywhere in the Godot API?

Reply:
That's how you convert an orthogonal matrix to quaternion. If it's not there it's a bug.

Reply:
If you mean the function itself it's operator Quat function in Basis. Can be exposed better for gdnative I guess

Reply:
But this is not exposed to scripts. We can't diverge like this from Godot's API. In the case we actually want to expose something different, it should probably be via extension methods (or making the type partial and separating it in multiple files, to make it easy to mantain).

Reply:
Again don't worry. I'll add those. They are supposed to be there